src/app/pages/apps/library-type-option/library-type-option.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormGroup, Validators, FormBuilder, UntypedFormControl  } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { ActivatedRoute } from '@angular/router';
+import { Subject, takeUntil } from 'rxjs';
+import { ILibraryTypeOption } from 'src/app/core/models/library-type-option';
+import { PortalService } from 'src/app/core/services/portal.service';
+
+@Component({
+  selector: 'vex-library-type-option',
+  templateUrl: './library-type-option.component.html',
+  styleUrls: ['./library-type-option.component.scss']
+})
+export class LibraryTypeOptionComponent implements OnInit, OnDestroy {
+  form: FormGroup; 
+  layoutCtrl = new UntypedFormControl('fullwidth');
+  statusLabel = 'Active';
+  createdBy: string;
+  createdAt: Date;
+  updatedBy: string;
+  updatedAt: Date;
+
+  get formControls() {
+    return {
+      type: this.form.get('type'),
+      code: this.form.get('code'),
+      description: this.form.get('description'),
+      status: this.form.get('status'),
+      remarks: this.form.get('remarks')
+    };
+  }
+
+  private _onDestroy$ = new Subject<void>();
+ 
+  constructor(
+    private _fb: FormBuilder,
+    private _route: ActivatedRoute,
+    private _portalService: PortalService,
+    private _snackBar: MatSnackBar
+  ) {
+    this.form = this._fb.group({
+      type: ['', [Validators.required]],
+      code: ['', [Validators.required]],
+      description: ['', [Validators.required]],
+      status: ['', [Validators.required]],
+      remarks: ['']
+      // ,isActive: [true, []]
```

Comment:
Remove this commented code if not needed.

Reply:
Kindly see update